pkg/signal/signal.go

```diff
@@ -159,12 +159,15 @@ func (s Signal) RandomSubset(r *rand.Rand, size int) Signal {
 	return ret
 }
 
-// FilterRaw returns a subset of original raw elements that coincides with the one in Signal.
-func (s Signal) FilterRaw(raw []uint32) []uint32 {
+// FilterRaw returns a subset of original raw elements that either coincides with the one in Signal,
+// or are not present in max.
+func (s Signal) FilterRaw(raw []uint32, max Signal) []uint32 {
```

Comment:
I think this function's signature has become too confusing now. Maybe we could at least make it

```
func FilterRaw(raw []uint32, ignore, alwaysTake Signal) []uint32 {
```

and call `signal.FilterRaw(raw, maxSignal, filterSignal)`.

Reply:
Done.
PTAL